openpype/hosts/cinema4d/plugins/publish/collect_instances.py

```diff
@@ -0,0 +1,185 @@
+import c4d
+import pyblish.api
+import json
+from openpype.hosts.cinema4d.api import lib
+
+
+
+class CollectInstances(pyblish.api.ContextPlugin):
+    """Gather instances by objectSet and pre-defined attribute
+
+    This collector takes into account assets that are associated with
+    an objectSet and marked with a unique identifier;
+
+    Identifier:
+        id (str): "pyblish.avalon.instance"
+
+    Limitations:
+        - Does not take into account nodes connected to those
+            within an objectSet. Extractors are assumed to export
+            with history preserved, but this limits what they will
+            be able to achieve and the amount of data available
+            to validators. An additional collector could also
+            append this input data into the instance, as we do
+            for `pype.rig` with collect_history.
```

Comment:
This is coming from Maya implementation - and I'm not sure is relevant to your code.
```suggestion
```